ols/app/models/config.py

```diff
@@ -27,15 +39,34 @@ def __init__(self, data: Optional[dict] = None):
             return
         self.name = data.get("name", None)
         self.url = data.get("url", None)
-        self.credential_path = data.get("credential_path", None)
+        self.credentials = data.get("credentials", None)
```

Comment:
we now have two fields named "credentials" one of which takes a filesystem path and the other which takes the actual credentials as the value.  (one is for provider creds, the other is for redis creds).

we should be consistent in aligning the naming and values, so please keep this name as `credentials_path` so it's clear to what sort of value is expected

Reply:
@bparees There're two perspectives. From the point of view of someone writing a config file, they are specifying a path to a file with credentials and `credentials_path` makes sense. From the point of view of someone writing code using config, there's just the value of credentials, so `credentials` makes sense. I tried to accommodate both perspectives in one of the past iterations where there were both fields in ProviderConfig and ModelConfig and I lifted the name of the file from `credentials_path` and populated `credentials' with the contents of that file. Or we can only use `credentials_path` in the config file, and have a `credentials` config attribute to access from the code, like so:
```
class ModelConfig(BaseModel):
...
    credentials: Optional[str] = None

   def __init__(self, data: Optional[dict] = None): 
       credentials_path = data.get("credentials_path", None)
        if credentials_path is not None:
            with open(credentials_path, mode="r") as f:
                self.credentials = f.read().rstrip()
...                
```

Reply:
@bparees On the Redis creds, don't we want to load them from a file as well? They are only mentioned in the sample config at https://gist.github.com/bparees/89419b57a54d52dfdfe5fa1506a7d174 but didn't make it to the code https://github.com/openshift/lightspeed-service/blob/aa8e5891849c69bdc7d0f1f0c81e4415fb9943bf/ols/app/models/config.py#L78

Reply:
> On the Redis creds, don't we want to load them from a file as well?

yes, i'd expect them to be mounted into the OLS pod on disk also and then referenced from the config yaml

Reply:
> Or we can only use credentials_path in the config file, and have a credentials config attribute to access from the code

this approach works for me